pkgs/jni/lib/src/jclass.dart

```diff
@@ -0,0 +1,154 @@
+  }
+}
+
+/// A thin wrapper over a [JFieldIDPtr] of an instance field.
+extension type JInstanceField._fromPointer(JFieldIDPtr pointer) {
+  JInstanceField._(JClass jClass, String name, String signature)
+      : pointer = using((arena) => Jni.accessors
+            .getFieldID(
+              jClass.reference.pointer,
+              name.toNativeChars(arena),
+              signature.toNativeChars(arena),
+            )
+            .fieldID);
+
+  DartT get<JavaT, DartT>(JObject object, JAccessible<JavaT, DartT> type) {
+    return type._instanceGet(object.reference.pointer, this as JFieldIDPtr);
+  }
+
+  void set<JavaT, DartT>(
+      JObject object, JAccessible<JavaT, DartT> type, DartT value) {
+    type._instanceSet(object.reference.pointer, this as JFieldIDPtr, value);
+  }
+}
+
+/// A thin wrapper over a [JFieldIDPtr] of an static field.
+extension type JStaticField._fromPointer(JFieldIDPtr pointer) {
+  JStaticField._(JClass jClass, String name, String signature)
+      : pointer = using((arena) => Jni.accessors
+            .getStaticFieldID(
+              jClass.reference.pointer,
+              name.toNativeChars(arena),
+              signature.toNativeChars(arena),
+            )
+            .fieldID);
+
+  DartT get<JavaT, DartT>(JClass jClass, JAccessible<JavaT, DartT> type) {
+    return type._staticGet(jClass.reference.pointer, this as JFieldIDPtr);
+  }
+
+  void set<JavaT, DartT>(
+      JObject object, JAccessible<JavaT, DartT> type, DartT value) {
+    type._staticSet(object.reference.pointer, this as JFieldIDPtr, value);
+  }
+}
+
+/// A thin wrapper over a [JMethodIDPtr] of an instance method.
+extension type JInstanceMethod._fromPointer(JMethodIDPtr pointer) {
+  JInstanceMethod._(
+    JClass jClass,
+    String name,
+    String signature,
+  ) : pointer = using((arena) => Jni.accessors
+            .getMethodID(
+              jClass.reference.pointer,
+              name.toNativeChars(arena),
+              signature.toNativeChars(arena),
+            )
+            .methodID);
+
+  /// Calls the instance method on [object] with the given arguments.
+  DartT call<JavaT, DartT>(
+    JObject object,
+    JCallable<JavaT, DartT> returnType,
+    List<dynamic> args,
+  ) {
+    return using((arena) => returnType._instanceCall(object.reference.pointer,
+        this as JMethodIDPtr, toJValues(args, allocator: arena)));
+  }
+}
+
+/// A thin wrapper over a [JMethodIDPtr] of a static mehtod.
+extension type JStaticMethod._fromPointer(JMethodIDPtr pointer) {
+  JStaticMethod._(
+    JClass jClass,
+    String name,
+    String signature,
+  ) : pointer = using((arena) => Jni.accessors
+            .getStaticMethodID(
+              jClass.reference.pointer,
+              name.toNativeChars(arena),
+              signature.toNativeChars(arena),
+            )
+            .methodID);
+
+  /// Calls the static method on [jClass] with the given arguments.
+  DartT call<JavaT, DartT>(
+    JClass jClass,
```

Comment:
Consider making this an actual object so the `jClass` doesn't have to be passed again.